python/python/res/simulator/robust_simulator.py

```diff
@@ -0,0 +1,121 @@
+            self.control_keys.append(key)
+
+        for key in results:
+            ens_config.addNode( EnkfConfigNode.create_gen_data( key , "{}_%d".format(key)))
+            self.result_keys.append(key)
+
+
+
+
+    def start(self, case_name, controls):
+        """Will start a robust simulation, returning a handle to query status and results.
+
+        The start method will submit simulations to the queue system and then
+        return a RobustContext handle which can be used to query for simulation
+        status and results. The @case_name argument should just be string which
+        will be used as name for the storage of these simulations in the
+        system. The @controls argument is the set of control values, and the
+        corresponding ID of the external realisation used for the simulations.
+        The @control argument must match the control argument used when the
+        simulator was instantiated. Assuming the following @control argument
+        was passed to simulator construction:
+
+             controls = {"cmode": ["Well","Group"], "order" : ["W1", "W2", "W3"]}
+
+        Then the following @controls argument can be used in the start method
+        to simulate four simulations:
+
+              [ (1, {"cmode" : [1 ,2], "order" : [2,2,5]}),
+                (1, {"cmode" : [1, 3], "order" : [2,2,7]}),
+                (1, {"cmode" : [1, 7], "order" : [2,0,5]}),
+                (2, {"cmode" : [1,-1], "order" : [2,2,1]})]
+
+        The first integer argument in the tuple is the realisation id, so this
+        simulation batch will consist of a total of four simulations, where the
+        three first are based on realisation 1, and the last is based on
+        realisation 2.
+        """
+        ens_config = self.res_config.ensemble_config
+        fsm = self.ert.getEnkfFsManager( )
+        fs = fsm.getFileSystem(case_name)
+
+        for sim_id, (geo_id, control_dict)  in enumerate(controls):
+            node_id = NodeId( 0, sim_id)
+            if len(control_dict) != len(self.control_keys):
+                raise ValueError("Not all keys supplied in controls")
+
+            for key in control_dict.keys():
+               config_node = ens_config[key]
+               ext_config = config_node.getModelConfig( )
+               values = control_dict[key]
+               if not len(values) == len(ext_config):
+                   raise ValueError("Wrong number of values for:%s" % key)
+
+               node = EnkfNode(config_node)
+               ext_node = node.as_ext_param( )
+               ext_node.set_vector(values)
+               node.save(fs, node_id)
+
+
+        mask = BoolVector( default_value = True, initial_size = len(controls) )
+        itr = 0
+        sim_context = RobustContext(self.result_keys, self.ert, fs, mask, itr)
+
+        for sim_id, (geo_id, control_dict)  in enumerate(controls):
+            sim_context.addSimulation(sim_id, geo_id)
+
+        return sim_context
```

Comment:
I was wondering, is there any limitation on "how often" `start()`can be called? I mean, if someone writes
```
rs = RobustSimulator(config, controls, results) 
c1 = rs.start('test1', controls_1)
c2 = rs.start('test_2, controls_2)
```
then `c1` and `c2` will share the same EnKFMain instance. Can this ever be a problem?

Reply:
Quite a lot of effort has gone into this question; I think the `EnkfMain` instance can be safely shared in this manner - although I would not bet my firstborn on it quite yet. When it comes to more peripheral aspects - in particular the folders where simulations are run this will create problems. But those problems are quite well contained.

Reply:
It may be worth adding a note about this in the documentation string of this method, which will be a warning to whoever tries to use this method, and a reminder for us that this should be checked.

An alternative could be to merge `__init__` and `start()`, so that we are sure each run gets a shiny new `EnKFMain`.

I am ok with both options, up to you ;)

Reply:
Comment added - would prefer the `EnkfMain`instance to stay around.